libinitializer/P2PInitializer.cpp

```diff
@@ -130,7 +139,8 @@ void P2PInitializer::initConfig(boost::property_tree::ptree const& _pt)
         m_p2pService->setStaticNodes(nodes);
         m_p2pService->setKeyPair(m_keyPair);
         m_p2pService->setP2PMessageFactory(messageFactory);
-
+        /// set compress handler
+        m_p2pService->setCompressHandler(compressHandler);
```

Comment:
should be a new messageFactory. According to config to decide which messageFactory is created.